### PR TITLE
Add automated release process

### LINKED ISSUE #41: Add release process and script

### PR DESCRIPTION
**Pull request by SeanTAllen**
Adds our standard GitHub based automated release process and associated
scripts.

Includes

- removal of CircleCI
- nightly releases to Cloudsmith
- tagged releases to Cloudsmith
- "latest" tag Docker images to DockerHub
- "release" and "X.Y.Z" tag Docker images to DockerHub
- changelog-bot support
- PR CI workflow
- Release process instructions

Closes #41

### CODE REVIEW

File: .ci-scripts/release/build-docker-images-on-release.bash [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+set -o nounset
+
+# We aren't validating TAG is in our x.y.z format but we could.
+# For now, TAG validating is left up to the configuration in
+# our GitHub workflow
+# Tag ref version: "refs/tags/1.0.0"
+# Version: "1.0.0"
+VERSION="${GITHUB_REF/refs\/tags\//}"
+
+# Build and push :VERSION tag e.g. ponylang/ponyup:0.32.1
```
Comment (Theodus) [inline .ci-scripts/release/build-docker-images-on-release.bash]: I think these should be updated (ponyup -> corral)

Comment (SeanTAllen) [reply .ci-scripts/release/build-docker-images-on-release.bash]: I was very intentional about not changes. It's an example, I don't want to have to change the files each time we install. 

Is there something other than ponyup to put there that can be generic across any application?

Comment (Theodus) [reply .ci-scripts/release/build-docker-images-on-release.bash]: I was about to say ponyc, but those versions are different. So this is fine

Comment (SeanTAllen) [reply .ci-scripts/release/build-docker-images-on-release.bash]: Ok.

**Pull request merged**